test/helpers/mock_server.rb

```diff
@@ -0,0 +1,57 @@
+# frozen_string_literal: true
+
+require 'webrick'
+
+module MockServer
+  extend self
+
+  def start(hostname:, port:)
+    tid = Thread.new do
+      start_server(hostname, port)
+    end
+    poll_until_ready(hostname: hostname, port: port)
+    puts "Created test server #{tid}, port: #{port}"
+    tid
+  end
+
+  def cleanup(tid)
+    Thread.kill(tid)
+    puts "Killed test server #{tid}."
+  end
+
+  def start_server(hostname, port)
+    server = TCPServer.new(hostname, port)
+    while (sock = server.accept)
+      begin
+        config = WEBrick::Config::HTTP
+        res = WEBrick::HTTPResponse.new(config)
+        req = WEBrick::HTTPRequest.new(config)
+        req.parse(sock)
+
+        response_code = req.path_info.delete("/")
+        response_code = 200 if response_code == ""
+
+        res.status = response_code
+        res.content_type = 'text/html'
+      rescue WEBrick::HTTPStatus::EOFError, WEBrick::HTTPStatus::BadRequest
+        res.status = 200
```

Comment:
Is 200 the most useful status in this case?

Reply:
Switched it out to return 400 in #235